cultivation/systems/arc_reactor/jarc_reactor/data/eval_data_prep.py

```
@@ -0,0 +1,308 @@
+def load_context_pairs(directory, context_map):
+    logger.info(f"Loading context pairs from '{directory}'...")
+    json_files = [f for f in os.listdir(directory) if f.endswith('.json')]
+    with ThreadPoolExecutor() as executor:
+        futures = {
+            executor.submit(load_context_pair, os.path.join(directory, filename), os.path.splitext(filename)[0], context_map): filename
+            for filename in json_files
+        }
+        for future in tqdm(as_completed(futures), total=len(futures), desc="Loading context pairs"):
+            future.result()  # Ensure any exceptions are raised
+
+def load_main_data_concurrently(directory, context_map, train_inputs, train_outputs, train_task_ids, train_context_pairs,
```

Pylint (3.3.7)

[refactor] Too many arguments (11/5)

(R0913)